src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerForV7.php

```diff
@@ -36,6 +36,8 @@ public function __construct(callable $sutFqcnResolver = null, $warningOnSutNotFo
 
     public function startTestSuite(TestSuite $suite): void
     {
-        $this->trait->startTest($suite);
+        foreach ($suite->tests() as $test) {
+            $this->trait->startTest($test);
+        }
```

Comment:
An alternate fix would be to implement `startTest` instead of `startTestSuite`, not sure why it wasn't done like this in #26089…

Reply:
And I thought yet another fix would be to let `startTrait` handle the test suite by completing the `if` test, but it doesn't work because `getAnnotations` is called in the trait, and the test suite does not have this method.

Reply:
> implement startTest instead of startTestSuite
let's do that?

Reply:
Sure! See #28495, pick whichever you prefer!

Reply:
Would be great to have feedback from @lyrixx as to why only the v7 listener uses `TestSuite` though